src/HotChocolate/Core/test/Types.Scalars.Tests/NegativeIntTypeTests.cs

```diff
@@ -0,0 +1,117 @@
+        [InlineData(typeof(NullValueNode), null, true)]
+        public void IsInstanceOfType_GivenValueNode_MatchExpected(
+            Type type,
+            object value,
+            bool expected)
+        {
+            // arrange
+            IValueNode valueNode = CreateValueNode(type, value);
+
+            // act
+            // assert
+            ExpectIsInstanceOfTypeToMatch<NegativeIntType>(valueNode, expected);
+        }
+
+        [Theory]
+        [InlineData(TestEnum.Foo, false)]
+        [InlineData(1d, false)]
+        [InlineData(-1, true)]
+        [InlineData(true, false)]
+        [InlineData("", false)]
+        [InlineData(null, true)]
+        [InlineData("foo", false)]
+        public void IsInstanceOfType_GivenObject_MatchExpected(object value, bool expected)
+        {
+            // arrange
+            // act
+            // assert
+            ExpectIsInstanceOfTypeToMatch<NegativeIntType>(value, expected);
+        }
+
+        [Theory]
+        [InlineData(typeof(IntValueNode), -12, -12)]
+        [InlineData(typeof(NullValueNode), null, null)]
+        public void ParseLiteral_GivenValueNode_MatchExpected(
+            Type type,
+            object value,
+            object expected)
+        {
+            // arrange
+            IValueNode valueNode = CreateValueNode(type, value);
+
+            // act
+            // assert
+            ExpectParseLiteralToMatch<NegativeIntType>(valueNode, expected);
+        }
+
+        [Theory]
+        [InlineData(typeof(EnumValueNode), TestEnum.Foo)]
+        [InlineData(typeof(FloatValueNode), 1d)]
+        [InlineData(typeof(IntValueNode), 0)]
+        [InlineData(typeof(IntValueNode), 1)]
+        [InlineData(typeof(BooleanValueNode), true)]
+        [InlineData(typeof(StringValueNode), "")]
+        public void ParseLiteral_GivenValueNode_ThrowSerializationException(Type type, object value)
+        {
+            // arrange
+            IValueNode valueNode = CreateValueNode(type, value);
+
+            // act
+            // assert
+            ExpectParseLiteralToThrowSerializationException<NegativeIntType>(valueNode);
+        }
+
+        [Theory]
+        [InlineData(typeof(IntValueNode), -12)]
+        [InlineData(typeof(NullValueNode), null)]
+        public void ParseValue_GivenObject_MatchExpectedType(Type type, object value)
+        {
+            // arrange
+            // act
+            // assert
+            ExpectParseValueToMatchType<NegativeIntType>(value, type);
+        }
+
+        [Theory]
+        [InlineData(TestEnum.Foo)]
+        [InlineData(1d)]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(true)]
+        [InlineData("")]
+        public void ParseLiteral_GivenObject_ThrowSerializationException(object value)
+        {
+            // arrange
+            // act
+            // assert
+            ExpectParseValueToThrowSerializationException<NegativeIntType>(value);
+        }
+    }
```

Comment:
I forgot something in my pr :) 

We also need to test the serialisation and deserialisation

For the positive int I used these test cases:

```csharp

        [Theory]
        [InlineData(1, 1)]
        [InlineData(int.MaxValue, int.MaxValue)]
        [InlineData(null, null)]
        public void Deserialize_GivenValue_MatchExpected(
            object resultValue,
            object runtimeValue)
        {
            // arrange
            // act
            // assert
            ExpectDeserializeToMatch<PositiveIntType>(resultValue, runtimeValue);
        }

        [Theory]
        [InlineData(TestEnum.Foo)]
        [InlineData(1d)]
        [InlineData(true)]
        [InlineData("foo")]
        [InlineData(int.MinValue)]
        [InlineData(-1)]
        [InlineData(0)]
        public void Deserialize_GivenValue_ThrowSerializationException(object value)
        {
            // arrange
            // act
            // assert
            ExpectDeserializeToThrowSerializationException<PositiveIntType>(value);
        }

        [Theory]
        [InlineData(1, 1)]
        [InlineData(int.MaxValue, int.MaxValue)]
        [InlineData(null, null)]
        public void Serialize_GivenObject_MatchExpectedType(
            object runtimeValue,
            object resultValue)
        {
            // arrange
            // act
            // assert
            ExpectSerializeToMatch<PositiveIntType>(runtimeValue, resultValue);
        }

        [Theory]
        [InlineData(TestEnum.Foo)]
        [InlineData(1d)]
        [InlineData(true)]
        [InlineData("foo")]
        [InlineData(int.MinValue)]
        [InlineData(-1)]
        [InlineData(0)]
        public void Serialize_GivenObject_ThrowSerializationException(object value)
        {
            // arrange
            // act
            // assert
            ExpectSerializeToThrowSerializationException<PositiveIntType>(value);
        }

```